lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,417 @@
+
+        self.provenance_log = self.output_irf_file.parent / (
+            self.name + ".provenance.log"
+        )
+
+    def start(self):
+
+        for particle_type, p in self.mc_particle.items():
+            self.log.info(f"Simulated {particle_type.title()} Events:")
+            p["events"], p["simulation_info"] = read_mc_dl2_to_pyirf(p["file"])
+            # Check to see if the data table is indeed a QTable
+            self.data_sel(p["events"])
+            self.log.info("Data Table checked to be a QTable")
+            if p["simulation_info"].viewcone.value == 0.0:
+                p["mc_type"] = "point-like"
+            else:
+                p["mc_type"] = "diffuse"
+                # For diffuse gamma using Proton Spectra for calculating event weights
+                if particle_type == "gamma":
+                    p["target_spectrum"] = IRFDOC_PROTON_SPECTRUM
+                    self.log.debug(
+                        "Proton spectrum used as target spectrum"
+                        " for MC diffuse gamma"
+                    )
+
+            self.log.debug(f"Simulated {p['mc_type']} {particle_type.title()} Events:")
+
+            p["simulated_spectrum"] = PowerLaw.from_simulation(
+                p["simulation_info"], 50 * u.hour
+            )
+            p["events"]["weight"] = calculate_event_weights(
+                p["events"]["true_energy"],
+                p["target_spectrum"],
+                p["simulated_spectrum"],
+            )
+
+            for prefix in ("true", "reco"):
+                k = f"{prefix}_source_fov_offset"
+                p["events"][k] = calculate_source_fov_offset(p["events"], prefix=prefix)
+            # calculate theta / distance between reco and assumed source position
+            p["events"]["theta"] = calculate_theta(
+                p["events"],
+                assumed_source_az=p["events"]["true_az"],
+                assumed_source_alt=p["events"]["true_alt"],
+            )
+            self.log.debug(p["simulation_info"])
+
+        gammas = self.mc_particle["gamma"]["events"]
+
+        self.log.info(f"Using fixed G/H cut of {self.data_sel.fixed_gh_cut}")
+
+        gammas = self.data_sel.filter_cut(gammas)
+        gammas = self.data_sel.tel_ids_filter(gammas)
+        gammas = self.data_sel.gh_cut(gammas)
+
+        # point_like = True for point like IRFs, False for Full Enclosure IRFs
+        if self.point_like:
+            gammas = self.data_sel.theta_cut(gammas)
+            gammas = self.data_sel.true_src_fov_offset_cut(gammas)
+
+        # Binning of parameters used in IRFs
+        true_energy_bins = self.data_bin.true_energy()
+        reco_energy_bins = self.data_bin.reco_energy()
+        migration_bins = self.data_bin.energy_migration()
+        source_offset_bins = self.data_bin.source_offset()
+
+        if self.mc_particle["gamma"]["mc_type"] == "point-like":
+            # Gammapy 0.18.2 needs offset bin centers for interpolation
```

Comment:
I don't understand this. You are only multiplying by degree. Shouldn't the `DataBinning` already return the correct units? What exactly is the workaround? And did you open an issue with gammapy for that bug?

Can you link this issue here so we can check once it is fixed?

Reply:
Ah yes, I forgot to put the units in `DataBinning` for these bins.

I have not yet opened an issue in gammapy, though I did a follow-up on a previous issue raised on the slack channel, but did not get a resolution on it. I think, we can first merge this and then try to solve this bug with gammapy.

Reply:
Yes, but you should open the issue first, so you can reference the issue here with the workaround.